### DISCUSSION TITLE
Kiali setup with Federated Prometheus - Multiple clusters

### DISCUSSION THREAD

**Discussion by aleksandermiszkiewicz** [kiali/kiali#6837] [Q&A]
Hey,

I'm working on Kiali configuration for multiple K8s clusters. I would like to know if Kiali is able to work with the Federated Prometheus which gathers metrics from multiple K8s clusters. Below is the simplified architecture which will help to understand by question:

![Kiali_prometheus drawio](https://github.com/kiali/kiali/assets/96211187/df830bcb-db34-495e-adfa-4f3a13997488)

So simplifying, I have three K8s clusters, each for one env, so DEV, QA, PROD. Each cluster has it own Istio mesh configured. Additionally each cluster has it's own Kiali deployed. 

Based  on the Istio documentation about [Prometheus production setup](https://istio.io/latest/docs/ops/best-practices/observability/#workload-level-aggregation-via-recording-rules) and one more very interested article about the [Kiali configuration with production scale prometheus] (https://medium.com/kialiproject/kiali-with-production-scale-prometheus-c53ddfa20570) I would like to setup Kiali according to the Option 3 from the article, so where 
* the Prometheus is deployed on each cluster (with short retention period). Kiali is pointing to this Prometheus as main url thanks of this  I have graph, service health, Istio metrics in Kiali
* the Main Prometheus is deployed on the seperated cluster (the Federation Prometheus setup) and will use recording rules. Additionally, Kiali will be configured with custom dashboard and will use `external_services.prometheus.custom_metrics_url`  for this. 

Does this setup will work with Kiali ? Can the Main Prometheus be deployed on separate cluster and gather metrics from multiple Istio's Prometheus or it need to be deployed in each cluster otherwise.

**Comment by jmazzitelli** [comment]
There is a setting that will help Kiali work with a single federated Prometheus that contains telemetry from multiple meshes - `query_scope`. There are three query_scope settings - one for [custom dashboards](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.custom_dashboards.prometheus.query_scope), one for the [main prometheus](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.prometheus.query_scope), and one for [tracing queries](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.tracing.query_scope)

It's just a set of labels (see the [example CR](https://kiali.io/docs/configuration/kialis.kiali.io/#example-cr)) but it allows you to tell Kiali to filter out metrics from Prometheus queries (and Jaeger queries) that do not belong to the mesh that Kiali is observing.

This `query_scope` is mentioned in the [multi-cluster docs page](https://kiali.io/docs/configuration/multi-cluster/). You will want to read that page, too.
